tests/test_enterprise/views/test_grant_data_sharing_permissions.py

```diff
@@ -347,19 +350,22 @@ def test_get_course_specific_consent_bad_api_response(
             course_id=course_id,
             enterprise_customer=enterprise_customer,
         )
-        response = self.client.get(
-            self.url + '?course_id=course-v1%3AedX%2BDemoX%2BDemo_Course&next=https%3A%2F%2Fgoogle.com'
-        )
-        assert response.status_code == 404
+        with mock.patch('enterprise.views.render') as mock_render:
+            mock_render.return_value = HttpResponse()
+            self.client.get(
+                self.url + '?course_id=course-v1%3AedX%2BDemoX%2BDemo_Course&next=https%3A%2F%2Fgoogle.com',
+                data={
+                    'enterprise_customer_uuid': enterprise_customer.uuid,
```

Comment:
seems like we are testing something slightly different if we are sending in the enterprise_customer_uuid here when we didn't before.... i suspect this has to do with the ordering of your checks related to whether this param exists and how we get the enterprise customer object (see above)

Reply:
okay so what happened here is that get_enterprise_customer_or_404 was throwing a value error on a blank string being handed to it throughout my work, so i ended up tweaking this and the other test you commented on. i didn't assume the test suite was correct.

that said, ultimately, i ended up changing utils.py to catch the value error and to return a http404, so I will change this test back